pkg/webhook/sidecarset/mutating/hash.go

```diff
@@ -40,6 +40,9 @@ func SidecarSetHash(sidecarSet *appsv1alpha1.SidecarSet) (string, error) {
 // SidecarSetHashWithoutImage calculates hash without sidecars's image
 func SidecarSetHashWithoutImage(sidecarSet *appsv1alpha1.SidecarSet) (string, error) {
 	ss := sidecarSet.DeepCopy()
+	for i := range ss.Spec.InitContainers {
+		ss.Spec.InitContainers[i].Image = ""
+	}
```

Comment:
init container cannot be updated inplace, so the  image of init container should not be munged

Reply:
what is munged? We do need to include the initContainers in the hash to distinguish sidecarset that are only different in their initContainer settings, right?

Reply:
hash without image is used for the sidecarset controller to determine whether inplace update is possible once sidecarset changes. init container cannot be changed inplace, so the value must be included in the SidecarSetHashWithoutImage, and cannot be munged (modified to empty string)

Reply:
yes, I get that point and the current PR also has that. However, on second thought, I don't know how can we simply skip the inplace update when both the container and other parts of the sidecarset are both changed.